cloudinit/sources/DataSourceOVF.py

```diff
@@ -259,21 +269,43 @@ def transport_iso9660(require_iso=True):
 
 
 def transport_vmware_guestinfo():
-    rpctool = "vmware-rpctool"
-    not_found = None
-    if not subp.which(rpctool):
-        return not_found
-    cmd = [rpctool, "info-get guestinfo.ovfEnv"]
+    is_vmtoolsd = False
+    rpctool = subp.which("vmware-rpctool")
+
+    if not rpctool:
+        rpctool = subp.which("vmtoolsd")
+        is_vmtoolsd = True
+
+    if not rpctool:
+        return None
+
+    try:
+        return transport_vmware_guestinfo_rpctool(rpctool, is_vmtoolsd)
+    except subp.ProcessExecutionError:
+        rpctool = subp.which("vmtoolsd")
+        is_vmtoolsd = True
+        return transport_vmware_guestinfo_rpctool(rpctool, is_vmtoolsd)
+
+
+def transport_vmware_guestinfo_rpctool(rpctool, is_vmtoolsd=False):
+    args = [rpctool]
+    if is_vmtoolsd:
+        args.append("--cmd")
+    args.append("info-get guestinfo.ovfEnv")
```

Comment:
rather than a space, these need to be two separate arguments in the list, correct?

Reply:
Are you asking about `--cmd` and `info-get guestinfo.ovfEnv` or `info-get` and `guestinfo.ovfEnv`? If the latter, `info-get guestinfo.ovfEnv` is one arugment bookended by quotes.

Reply:
It was the latter. Sounds good!